src/drivers/amd/renoir/acp_sp_dai.c

```diff
@@ -70,11 +118,21 @@ static int spdai_get_hw_params(struct dai *dai,
 			      struct sof_ipc_stream_params *params,
 			      int dir)
 {
-	/* SP DAI currently supports only these parameters */
-	params->rate = ACP_DEFAULT_SAMPLE_RATE;
-	params->channels = ACP_DEFAULT_NUM_CHANNELS;
-	params->buffer_fmt = SOF_IPC_BUFFER_INTERLEAVED;
-	params->frame_fmt = SOF_IPC_FRAME_S16_LE;
+	struct acp_pdata *acpdata = dai_get_drvdata(dai);
+
+	if (dir == DAI_DIR_PLAYBACK) {
+		/* SP DAI currently supports only these parameters */
+		params->rate = ACP_DEFAULT_SAMPLE_RATE;
+		params->channels = acpdata->params.tdm_slots;
+		params->buffer_fmt = SOF_IPC_BUFFER_INTERLEAVED;
+		params->frame_fmt = SOF_IPC_FRAME_S16_LE;
+	} else if (dir == DAI_DIR_CAPTURE) {
+		/* SP DAI currently supports only these parameters */
+		params->rate = ACP_DEFAULT_SAMPLE_RATE;
+		params->channels = 2;
```

Comment:
Is this saying that you can have more than 2 slots on playback and only 2 on capture?

Reply:
yes, more than two slots for playback and two for capture.

Reply:
Is this saying that you have a dedicated bclk/frame sync on playback and capture, so a 6-pin interface then?

The Intel TDM only supports 4pins with shared bclk/frame_sync, so the number of slots has to be identical on playback and capture. You should comment on this difference to avoid misunderstanings.